python/python/glide/async_commands/standalone_commands.py

```diff
@@ -665,3 +665,66 @@ async def random_key(self) -> Optional[str]:
             Optional[str],
             await self._execute_command(RequestType.RandomKey, []),
         )
+
+    async def scan(
+        self,
+        cursor: str,
+        match: Optional[str] = None,
+        count: Optional[int] = None,
+        type: Optional[ObjectType] = None,
+    ) -> List[Union[str, List[str]]]:
+        """
+        Incrementally iterate over a collection of keys.
+        SCAN is a cursor based iterator. This means that at every call of the command,
+        the server returns an updated cursor that the user needs to use as the cursor argument in the next call.
+        An iteration starts when the cursor is set to "0", and terminates when the cursor returned by the server is "0".
+
+        A full iteration always retrieves all the elements that were present
+        in the collection from the start to the end of a full iteration.
+        Elements that were not constantly present in the collection during a full iteration, may be returned or not.
+
+        See https://valkey.io/commands/scan for more details.
+
+        Args:
+            cursor (str): The cursor used for iteration. For the first iteration, the cursor should be set to "0".
+              Using a non-zero cursor in the first iteration,
+              or an invalid cursor at any iteration, will lead to undefined results.
+            match (Optional[str]): A pattern to match keys against,
+                for example, "key*" will return all keys starting with "key".
+            count (Optional[int]): The number of keys to return per iteration.
+                The number of keys returned per iteration is not guaranteed to be the same as the count argument.
+                the argument is used as a hint for the server to know how many "steps" it can use to retrieve the keys.
+                The default value is 10.
+            type (ObjectType): The type of object to scan for: STRING, LIST, SET, HASH, ZSET.
+
+        Returns:
+            List[Union[str, List[str]]]: A List containing the next cursor value and a list of keys,
+                formatted as ["23", [key1, key2, ...]]
+
+        Examples:
+            >>> result = await client.scan("0")
+                print(result) #["17", ['key1', 'key2', 'key3', 'key4', 'key5', 'set1', 'set2', 'set3']]
+                first_cursor_result = result[0]
+                result = await client.scan(first_cursor_result)
+                print(result) #["349", ['key4', 'key5', 'set1', 'hash1', 'zset1', 'list1', 'list2',
+                                        'list3', 'zset2', 'zset3', 'zset4', 'zset5', 'zset6']]
+                result = await client.scan(result[0])
+                print(result) #["0", ['key6', 'key7']]
+
+            >>> result = await client.scan(first_cursor_result, match="key*", count=2)
```

Comment:
Can you add a short comment explaining why/what it means that you use an old cursor?

Reply:
or what we expect that it would return